dbt/adapters/trino/connections.py

```diff
@@ -81,6 +81,48 @@ def _connection_keys(self):
     def trino_auth() -> Optional[trino.auth.Authentication]:
         pass
 
+    def __post_init__(self):
+        if TrinoCredentials._HTTP_CLIENT_PATCHED:
+            return
+        if (self.patch_http_client_header_limit):
+            TrinoCredentials.patch_http_client()
+            TrinoCredentials._HTTP_CLIENT_PATCHED = True
+
+    @classmethod
+    def patch_http_client(cls):
+        """
+        Python's http client has a hardcoded limit of 65536 bytes for a header
+        line.
+
+        When executing a prepared statement with a large number of parameters,
+        you might encounter following error:
+
+        ```requests.exceptions.ConnectionError: ('Connection aborted.',
+        LineTooLong('got more than 65536 bytes when reading header line'))```
+
+        We work around this limit by executing this code before running any
+        queries.  Note that overriding Python's http client header handling may
+        also affect any other applications running within that same Python
+        environment, but because it's executed as part of a dbt execution. This
+        is probably less likely to occur.
+        """
+
+        import http.client
+        import email.parser
+        from http.client import HTTPMessage, _MAXLINE
+
+        def parse_headers(fp, _class=HTTPMessage):
+            headers = []
+            while True:
+                line = fp.readline(_MAXLINE + 1)
+                headers.append(line)
+                if line in (b'\r\n', b'\n', b''):
+                    break
+            header_string = b''.join(headers).decode('iso-8859-1')
```

Comment:
why Latin-1 encoding?

Reply:
That's the encoding according the [http spec](https://www.rfc-editor.org/rfc/rfc7230#section-3.2.4). 

> Historically, HTTP has allowed field content with text in the
>    ISO-8859-1 charset [[ISO-8859-1](https://www.rfc-editor.org/rfc/rfc7230#ref-ISO-8859-1)], supporting other charsets only
>    through use of [[RFC2047](https://www.rfc-editor.org/rfc/rfc2047)] encoding.  In practice, most HTTP header
>    field values use only a subset of the US-ASCII charset [[USASCII](https://www.rfc-editor.org/rfc/rfc7230#ref-USASCII)].
>    Newly defined header fields SHOULD limit their field values to
>    US-ASCII octets.  A recipient SHOULD treat other octets in field
>    content (obs-text) as opaque data.